picard/util/pipe.py

```diff
@@ -0,0 +1,206 @@
+    import win32pipe  # type: ignore
+    import win32file  # type: ignore
+    from pywintypes import error as WinApiError  # type: ignore
+
+    PLATFORM = "windows"
+
+
+class Pipe:
+    NO_RESPONSE_MESSAGE: str = "No response from FIFO"
+    NOT_FOUND_MESSAGE: str = "FIFO doesn't exist"
+    MESSAGE_TO_IGNORE: str = "Ignore this message, just testing the pipe"
+
+    def __init__(self, app_name: str, app_version: str, args=[]):
+        self.__app_name: str = app_name
+        self.__app_version: str = app_version
+        self.__platform: str = PLATFORM
+
+        self.path: str = self.__generate_filename()
+
+        self.is_pipe_owner: bool = False
+
+        # test if pipe is listened to even if no args provided
+        if type(args) == list:
+            if len(args) == 0:
+                args.append(self.MESSAGE_TO_IGNORE)
+        else:
+            raise ValueError("args argument MUST be a list")
+
+        if self.__platform == "windows":
+            for arg in args:
+                if not self.send_to_pipe(arg):
+                    self.is_pipe_owner = True
+                    break
+        else:
+            if self.__pipe_exists():
+                for arg in args:
+                    if not self.send_to_pipe(arg):
+                        os.unlink(self.path)
+                        self.__create_unix_pipe()
+                        break
+            else:
+                self.__create_unix_pipe()
+
+    def __pipe_exists(self) -> bool:
+        return os.path.exists(self.path)
+
+    def __generate_filename(self) -> str:
+        prefix: str = ""
+        username: str = os.getlogin()
+        if self.__platform == "windows":
+            prefix = "\\\\.\\pipe\\"
+        else:
+            prefix = "/tmp/"
+
+        return f"{prefix}{self.__app_name}_v{self.__app_version}_{username}_pipe_file"
+
+    def __create_unix_pipe(self) -> None:
+        os.mkfifo(self.path)
+        self.is_pipe_owner = True
+
+    def __win_sender(self, message: str) -> bool:
+        pipe = win32pipe.CreateNamedPipe(
+            self.path,
+            win32pipe.PIPE_ACCESS_DUPLEX,
+            win32pipe.PIPE_TYPE_MESSAGE | win32pipe.PIPE_WAIT,
+            1, 65536, 65536,
+            300,
+            None)
+        try:
+            win32pipe.ConnectNamedPipe(pipe, None)
+            win32file.WriteFile(pipe, str.encode(message))
+        finally:
+            win32file.CloseHandle(pipe)
+
+        return True
+
+    def __unix_sender(self, message: str) -> bool:
+        with open(self.path, 'a') as fifo:
+            fifo.write(message)
+        return True
+
+    def send_to_pipe(self, message: str, timeout_secs: float = 1.5) -> bool:
+        __pool = concurrent.futures.ThreadPoolExecutor()
+        sender = None
+        if self.__platform == "windows":
+            sender = __pool.submit(self.__win_sender, message)
+        else:
+            sender = __pool.submit(self.__unix_sender, message)
```

Comment:
```python

if IS_WIN:
    sender_method = self.__win_sender
else:
    sender_method = self.__unix_sender

sender = __pool.submit(sender_method, message)
```

Reply:
Great idea to avoid the boilerplate, gonna change it asap